src/components/cards/CardNameModal/hook/useCardNameModal.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from "react";
+
+import { CARD_NAMES, CardName } from "@/constants/variables";
+
+const useCardNameModal = (initialState = false) => {
+  const [isShow, setShowStatus] = useState(initialState);
+  const [cardNameList, setCardNameList] = useState<CardName[]>([]);
+  const [selectedCardName, setSelectedCardName] = useState<CardName>({
+    name: "",
+    color: "#94DACD",
+  });
+
+  useEffect(() => {
+    setCardNameList(CARD_NAMES);
+  }, []);
+
+  useEffect(() => {
+    if (!selectedCardName) return;
+
+    setSelectedCardName(CARD_NAMES[0]);
+  }, [cardNameList]);
+
+  const handleCardNameModalClose = () => {
+    setShowStatus(false);
+  };
+
+  const handleCardNameModalOpen = () => {
+    setShowStatus(true);
+  };
+
+  const handleCardNameSelect = (name: string, color: string) => {
+    setSelectedCardName({ name, color });
+    setShowStatus(false);
+  };
```

Comment:
```suggestion
  const closeModal = () => {
    setShowStatus(false);
  };

  const openModal = () => {
    setShowStatus(true);
  };

  const handleCardNameSelect = (name: string, color: string) => {
    setSelectedCardName({ name, color });
    closeModal();
  };
```
useCardNameModal 내부에 정의된 함수라 네이밍 위처럼 바꿔줘도 충분히 의도가 전달될 것 같아요~
내부 함수 네이밍은 간결한게 좋은 것 같아서 위처럼 해보면 어떨까요?